update RLS and rustfmt

calebcartwright opened the pull request:
Fixes #81582 and fixes #81583

r? @Xanewok

I was originally surprised by the size of lockfile diff, though after looking at the RLS changes it makes a bit more sense to me now

rust-highfive commented:
:warning: **Warning** :warning:

* These commits modify **submodules**.

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/1835513468) [(plain)](https://github.com/rust-lang/rust/commit/4d1b91eecfd46c2a1fb9ebfb687d2927d7933ee7/checks/1835513468/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#####################################################                     74.0%
######################################################################## 100.0%
extracting /checkout/obj/build/cache/2021-01-28/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
    Updating crates.io index
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
   Compiling merge v0.1.0
   Compiling toml v0.5.7
    Finished dev [unoptimized + debuginfo] target(s) in 47.57s
Checking stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking getopts v0.2.21
    Checking test v0.0.0 (/checkout/library/test)
    Finished release [optimized] target(s) in 29.15s
Checking stage0 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking rustc_driver v0.0.0 (/checkout/compiler/rustc_driver)
    Checking rustc-main v0.0.0 (/checkout/compiler/rustc)
    Finished release [optimized] target(s) in 1m 05s
Checking stage0 rustdoc artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking cranelift-jit v0.69.0 (https://github.com/bytecodealliance/wasmtime/?branch=main#986b5768)
    Checking rustc_codegen_cranelift v0.1.0 (/checkout/compiler/rustc_codegen_cranelift)
    Finished release [optimized] target(s) in 26.92s
Checking stage0 clippy artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking cargo_metadata v0.12.0
    Checking clippy_lints v0.1.51 (/checkout/src/tools/clippy/clippy_lints)
    Finished release [optimized] target(s) in 24.88s
Checking stage0 bootstrap artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
    Checking lazy_static v1.4.0
    Checking cfg-if v0.1.10
   Compiling log v0.4.11
    Checking regex-syntax v0.6.22
---
configure: rust.channel         := nightly
configure: rust.debug-assertions := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
configure: 
---
downloading https://static.rust-lang.org/dist/2021-01-28/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz

######################################################################## 100.0%
extracting /checkout/obj/build/cache/2021-01-28/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
   Compiling unicode-xid v0.2.1
   Compiling syn v1.0.38
   Compiling version_check v0.9.1
   Compiling autocfg v1.0.0
---
Hugetlb:               0 kB
DirectMap4k:      206784 kB
DirectMap2M:     3987456 kB
DirectMap1G:    56623104 kB
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
Ensuring the YAML anchors in the GitHub Actions config were expanded
Building stage0 tool expand-yaml-anchors (x86_64-unknown-linux-gnu)
Building stage0 tool expand-yaml-anchors (x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
   Compiling expand-yaml-anchors v0.1.0 (/checkout/src/tools/expand-yaml-anchors)
    Finished release [optimized] target(s) in 6.73s
Build completed successfully in 0:00:07
+ python3 ../x.py check --target=i686-pc-windows-gnu --host=i686-pc-windows-gnu --all-targets
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
Checking stage0 std artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
Checking stage0 std artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
    Checking core v0.0.0 (/checkout/library/core)
   Compiling libc v0.2.85
   Compiling std v0.0.0 (/checkout/library/std)
   Compiling compiler_builtins v0.1.39
---
    Checking getopts v0.2.21
    Checking test v0.0.0 (/checkout/library/test)
    Finished release [optimized] target(s) in 29.31s
Checking stage0 std test/bench/example targets (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking std v0.0.0 (/checkout/library/std)
    Checking alloc v0.0.0 (/checkout/library/alloc)
    Finished release [optimized] target(s) in 21.53s
Checking stage0 compiler artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking rustc_driver v0.0.0 (/checkout/compiler/rustc_driver)
    Checking rustc-main v0.0.0 (/checkout/compiler/rustc)
    Finished release [optimized] target(s) in 1m 06s
Checking stage0 rustdoc artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
    Checking cfg-if v0.1.10
    Checking lazy_static v1.4.0
   Compiling proc-macro2 v1.0.19
   Compiling unicode-xid v0.2.1
---
warning: 1 warning emitted

    Finished release [optimized] target(s) in 20.37s
Checking stage0 clippy artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking clippy_lints v0.1.51 (/checkout/src/tools/clippy/clippy_lints)
    Checking rustc-workspace-hack v1.0.0 (/checkout/src/tools/rustc-workspace-hack)
    Finished release [optimized] target(s) in 27.15s
Checking stage0 bootstrap artifacts (x86_64-unknown-linux-gnu -> i686-pc-windows-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
    Checking serde_json v1.0.59
    Finished release [optimized] target(s) in 16.48s
Build completed successfully in 0:03:19
+ python3 ../x.py build --stage 0 src/tools/build-manifest
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
Building stage0 tool build-manifest (x86_64-unknown-linux-gnu)
Building stage0 tool build-manifest (x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
   Compiling build-manifest v0.1.0 (/checkout/src/tools/build-manifest)
    Finished release [optimized] target(s) in 17.02s
Build completed successfully in 0:00:17
+ python3 ../x.py test --stage 0 src/tools/compiletest
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
Building stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
Building stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
   Compiling core v0.0.0 (/checkout/library/core)
   Compiling libc v0.2.85
   Compiling std v0.0.0 (/checkout/library/std)
   Compiling compiler_builtins v0.1.39
---
   Compiling getopts v0.2.21
   Compiling test v0.0.0 (/checkout/library/test)
    Finished release [optimized] target(s) in 1m 09s
Copying stage0 std from stage0 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu / x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
test result: ok. 24 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.08s

Build completed successfully in 0:01:31
+ python3 ../x.py test --stage 2 src/tools/tidy
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
Building stage0 tool tidy (x86_64-unknown-linux-gnu)
Building stage0 tool tidy (x86_64-unknown-linux-gnu)
warning: Patch `backtrace v0.3.55 (/checkout/library/backtrace)` was not used in the crate graph.
Check that the patched package version and available features are compatible
with the dependency requirements. If the patch has a different version from
what is locked in the Cargo.lock file, run `cargo update` to use the new
version. This may also occur with an optional dependency that is not enabled.
---
Found 436 error codes
Found 0 error codes with no tests
Done!
* 317 features
tidy error: could not find allowed package `backtrace`
Remove from PERMITTED_DEPENDENCIES list if it is no longer used.
some tidy checks failed

command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/tidy" "/checkout" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "/checkout/obj/build"
expected success, got: exit code: 1

```

</details>

calebcartwright commented on Cargo.lock:
```diff
@@ -5741,3 +5578,7 @@ checksum = "39f0c922f1a334134dc2f7a8b67dc5d25f0735263feec974345ff706bcf20b0d"
  "linked-hash-map",
 ]
+
+[[patch.unused]]
+name = "backtrace"
+version = "0.3.55"
```
Since RLS has removed the backtrace dep, this is now coming from

https://github.com/rust-lang/rust/blob/dab3a80f235ed86d64d002e58b35adc3664e21a4/Cargo.toml#L109-L113

I wasn't sure if that entry should be removed from the root Cargo.toml given the presence of the submodule and a few other references (looked like the RA submodule has it, but feature gated) so left in for now and just removed it from the permitted dep lists in the tidy check to get rid of the error.

Xanewok replied:
I think it's fine to leave it for now, especially if RA still uses it conditionally (I assume we dist it without this feature enabled?).

cc @alexcrichton just in case (https://github.com/rust-lang/rust/commit/06d565c967bfb7c6ff52a991bbe47b4a2a25de3e)

alexcrichton replied:
If nothing in rust-lang/rust is using the backtrace crate then this should be fine to remove. I thought Cargo at least would be using it through `anyhow`, but I was mistaken.

I think everything is using the dependency through the standard library now so it should be safe to remove.

calebcartwright replied:
> I assume we dist it without this feature enabled?

Actually I guess it's completely disabled (at least at the moment)

https://github.com/rust-analyzer/rust-analyzer/blob/80ab753d7e0bf59b81df317d6ddda43cb919ec83/crates/stdx/Cargo.toml#L12-L19
```toml
[dependencies]
backtrace = { version = "0.3.44", optional = true }
always-assert = { version = "0.1.2", features = ["log"] }
# Think twice before adding anything here

[features]
# Uncomment to enable for the whole crate graph
# default = [ "backtrace" ]
```

> I think everything is using the dependency through the standard library now so it should be safe to remove.

SGTM :+1: if something changes in that RA crate or elsewhere which brings backtrace back then the patch can be restored accordingly

calebcartwright replied:
Looks like bors will get to this before I get back to my dev machine. If so will open a followup PR to remove the patch

calebcartwright commented:
It's unclear to me why the other two jobs were cancelled, seemingly mid-run with everything succeeding up to that point :man_shrugging:

Xanewok commented:
Looks like an instance of https://github.com/rust-lang/rust/pull/81783.

I think it's worth getting it into the queue:

@bors r+ p=1 (sweeping lockfile changes)

bors commented:
:pushpin: Commit d43ffff3e6b3722a3f773164d231b405481aa3e9 has been approved by `Xanewok`

<!-- @bors r=Xanewok d43ffff3e6b3722a3f773164d231b405481aa3e9 -->
<!-- homu: {"type":"Approved","sha":"d43ffff3e6b3722a3f773164d231b405481aa3e9","approver":"Xanewok"} -->

bors commented:
:hourglass: Testing commit d43ffff3e6b3722a3f773164d231b405481aa3e9 with merge 0fcf134521aba2dbfa8344e7e470f348c24634bc...
<!-- homu: {"type":"BuildStarted","head_sha":"d43ffff3e6b3722a3f773164d231b405481aa3e9","merge_sha":"0fcf134521aba2dbfa8344e7e470f348c24634bc"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/1840717718) [(plain)](https://github.com/rust-lang-ci/rust/commit/0fcf134521aba2dbfa8344e7e470f348c24634bc/checks/1840717718/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

The actual stderr differed from the expected stderr.
Actual stderr saved to /tmp/compiletestbkI1t7/panic/catch_panic.stderr
To update references, run this command from build directory:
tests/run-pass/update-references.sh '/tmp/compiletestbkI1t7' 'panic/catch_panic.rs'
error: 1 errors occurred comparing output.
status: exit code: 0
status: exit code: 0
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools-bin/miri" "tests/run-pass/panic/catch_panic.rs" "-L" "/tmp/compiletestbkI1t7" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/tmp/compiletestbkI1t7/panic/catch_panic.stage-id" "-A" "unused" "--edition" "2018" "-Astable-features" "--sysroot" "/home/user/.cache/miri/HOST" "-Zmiri-symbolic-alignment-check" "-L" "/tmp/compiletestbkI1t7/panic/catch_panic.stage-id.aux"
------------------------------------------

------------------------------------------
stderr:
---
Verifying status of edition-guide...
Verifying status of rls...
This PR updated 'src/tools/rls', verifying if status is 'test-pass'...

We detected that this PR updated 'rls', but its tests failed.

If you do intend to update 'rls', please check the error messages above and
commit another update.

If you do NOT intend to update 'rls', please ensure you did not accidentally
change the submodule at 'src/tools/rls'. You may ask your reviewer for the
proper steps.
{"edition-guide":"test-pass","rls":"build-fail","nomicon":"test-pass","rustbook":"test-fail","cargo-miri":"test-fail","embedded-book":"test-pass","reference":"test-pass","rust-by-example":"test-pass","book":"test-pass","rustfmt":"build-fail","miri":"test-fail"}failed to run: /checkout/obj/build/bootstrap/debug/bootstrap test --stage 2 check-tools
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/1840959862)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/1840959862","builder_name":"checks-actions"} -->

calebcartwright commented:
Looks like we got caught between the stabilization of `int_bits_const` and the subsequent reversion in #81727. Will have to restart the auto publish crate update cycle

bors commented:
:umbrella: The latest upstream changes (presumably #81132) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#conflicts).

calebcartwright commented:
Rebased and updated to the v705 crate versions :crossed_fingers: I opted to keep the removal of the `backtrace` references in a separate commit on the off chance it needs to be reverted, but please lmk if you think everything should be squashed into one.

Xanewok commented:
Looks good, thanks for doing this!

@bors r+

bors commented:
:pushpin: Commit 0a47a38fd0099f654a1cd38e1de6bbb641863f13 has been approved by `Xanewok`

<!-- @bors r=Xanewok 0a47a38fd0099f654a1cd38e1de6bbb641863f13 -->
<!-- homu: {"type":"Approved","sha":"0a47a38fd0099f654a1cd38e1de6bbb641863f13","approver":"Xanewok"} -->

bors commented:
:hourglass: Testing commit 0a47a38fd0099f654a1cd38e1de6bbb641863f13 with merge 218bf8d7657e1aadf6f499651078f3710df20c7b...
<!-- homu: {"type":"BuildStarted","head_sha":"0a47a38fd0099f654a1cd38e1de6bbb641863f13","merge_sha":"218bf8d7657e1aadf6f499651078f3710df20c7b"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/1871355116)
Approved by: Xanewok
Pushing 218bf8d7657e1aadf6f499651078f3710df20c7b to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Xanewok","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/1871355116"},"merge_sha":"218bf8d7657e1aadf6f499651078f3710df20c7b"} -->

rust-highfive commented:
📣 Toolstate changed by rust-lang/rust#81768!

Tested on commit rust-lang/rust@218bf8d7657e1aadf6f499651078f3710df20c7b.
Direct link to PR: <https://github.com/rust-lang/rust/pull/81768>

🎉 rls on windows: build-fail → test-pass (cc @Xanewok).
🎉 rls on linux: build-fail → test-pass (cc @Xanewok).
🎉 rustfmt on windows: build-fail → test-pass (cc @topecongiro @calebcartwright).
🎉 rustfmt on linux: build-fail → test-pass (cc @topecongiro @calebcartwright).
